lib/middleware.js

```diff
@@ -218,9 +218,14 @@ module.exports = function(self, options) {
         }
       }
 
-      if (self.prefixes && self.prefixes[self.liveify(req.locale)] && (req.url === '/')) {
+      if (self.prefixes && self.prefixes[self.liveify(req.locale)] && req.url.match(/^\/(\?|$)/)) {
         // Redirect to home page of appropriate locale
-        return res.redirect(self.prefixes[self.liveify(req.locale)] + '/');
+        let newUrl = self.prefixes[self.liveify(req.locale)] + '/';
+        const matches = req.url.match(/^\/(\?.*)$/);
```

Comment:
This includes the `/`, doesn't it? So you would end up with `new.url//?query=string`.

Reply:
Nope, not that I can see:

```
const urls = [
  '/', '/foo', '/foo/bar', '/?bar=wizzle'
];

for (const url of urls) {
  if (url.match(/^\/(\?|$)/)) {
    let newUrl = '/prefixed/';
    const matches = url.match(/^\/(\?.*)$/);
    if (matches) {
      newUrl += matches[1];
    }
    console.log(url + ' is now ' + newUrl);
  } else {
    console.log('just ' + url);
  }
}
```

Behaves properly.